clevertap-core/src/main/java/com/clevertap/android/sdk/db/DBAdapter.kt

```diff
@@ -132,26 +132,68 @@ internal class DBAdapter(context: Context, config: CleverTapInstanceConfig) {
     }
 
     @Synchronized
-    fun fetchUserProfileById(id: String?): JSONObject? {
-        if (id == null) {
-            return null
+    fun fetchUserProfilesByAccountId(accountId: String?): Map<String,JSONObject> {
+        if (accountId == null) {
+            return emptyMap()
         }
+
+        val profiles = mutableMapOf<String, JSONObject>()
         val tName = USER_PROFILES.tableName
-        var profileString: String? = null
+
         try {
-            dbHelper.readableDatabase.query(tName, null, "${Column.ID} = ?", arrayOf(id), null, null, null)
+            dbHelper.readableDatabase.query(tName, null, "${Column.ID} = ?", arrayOf(accountId), null, null, null)
                 ?.use { cursor ->
-                    if (cursor.moveToFirst()) {
-                        val dataIndex = cursor.getColumnIndex(Column.DATA)
-                        if (dataIndex >= 0) {
-                            profileString = cursor.getString(dataIndex)
+                    val dataIndex = cursor.getColumnIndex(Column.DATA)
+                    val deviceId = cursor.getColumnIndex(Column.DEVICE_ID)
+                    if (dataIndex >= 0) {
+                        while (cursor.moveToNext()) {
+                            val profileString = cursor.getString(dataIndex)
+                            val deviceIdString = cursor.getString(deviceId)
+                            profileString?.let {
+                                try {
+                                    val jsonObject = JSONObject(it)
+                                    profiles.put(deviceIdString, jsonObject)
+                                } catch (e: JSONException) {
+                                    logger.verbose("Error parsing JSON for profile", e)
+                                }
+                            }
                         }
                     }
                 }
         } catch (e: SQLiteException) {
             logger.verbose("Could not fetch records out of database $tName.", e)
         }
 
+        return profiles
+    }
+
+    @Synchronized
+    fun fetchUserProfileByAccountIdAndDeviceID(accountId: String?, deviceId: String?): JSONObject? {
+        if (accountId == null || deviceId == null) {
+            return null
+        }
+        val tName = USER_PROFILES.tableName
+        var profileString: String? = null
+        try {
+            dbHelper.readableDatabase.query(
+                tName,
+                null,
+                "${Column.ID} = ? AND ${Column.DEVICE_ID} = ?",
+                arrayOf(accountId, deviceId),
+                null,
+                null,
+                null
+            )?.use { cursor ->
+                if (cursor.moveToFirst()) {
```

Comment:
make sure `cursor.moveToFirst()` does not throw exception if no rows are returned.

Reply:
False is returned if cursor is empty